packages/components/src/popover/index.js

```diff
@@ -61,7 +61,13 @@ class Popover extends Component {
 	componentDidMount() {
 		this.toggleWindowEvents( true );
 		this.refresh();
-		this.focus();
+		/*
+		 * Without the setTimeout, the dom node is not being focused. Related:
+		 * https://stackoverflow.com/questions/35522220/react-ref-with-focus-doesnt-work-without-settimeout-my-example
+		 */
+		setTimeout( () => {
```

Comment:
`setTimeout` in a component is a code smell. And in the very worst cases where we decide it's needed, it should always be unset when the component unmounts. This is what's intended with the [`withSafeTimeout` higher-order component](https://github.com/WordPress/gutenberg/tree/master/packages/compose/src/with-safe-timeout). I'm aware you didn't introduce this. I'll see if I can refactor it simply while we're here.

Reply:
Yep I know. I've tried `withSafeTimeout` but it didn't work. I guess it can't with popovers.

Reply:
Yes, always. `setTimeout`, even with a duration of `0`, occurs only after the current execution stack completes, meaning that anything else after the `setTimeout` is scheduled which would cause the component to become unmounted (which occurs very frequently in a React context) would cause an error to occur when the callback of the `setTimeout` assumes that the component still exists.

Demo: https://codepen.io/aduth/pen/ejyyyN?editors=1111

Reply:
OK thanks.